src/lib/build-server.ts

```diff
@@ -42,44 +48,36 @@ export async function buildServer({
 		logger: envToLogger[NODE_ENV] ?? true,
 		disableRequestLogging: !(NODE_ENV === "development" || NODE_ENV === "test"),
 	});
-	// eslint-disable-next-line @typescript-eslint/ban-ts-comment
-	// @ts-expect-error
-	await server.register(fastifySwagger, {
-		mode: "dynamic",
-		openapi: {
-			info: {
-				title: String,
-				description: String,
-				version: String,
-			},
-			externalDocs: Object,
-
-			components: Object,
-		},
-	});
+	server.register(cors, corsConfiguration);
 
+	server.register(fastifySwagger, swaggerOptions);
 	// Set rate limit
-	await server.register(import("@fastify/rate-limit"), {
+	server.register(import("@fastify/rate-limit"), {
 		max: 30,
 		timeWindow: "1 minute",
 	});
-
-	registerDocumentationRoute(server);
-	registerRootRoute(server);
-	registerHealthRoute(server);
-	registerCountDocumentsRoute(server);
+	server.register(fastifySwaggerUi, documentationRouteOptions);
+	server.register(rootRoute);
+	server.register(healthRoute, { prefix: "/health" });
+	// FIXME: We register at /processed_documents and add one handler for /count. Why?
```

Comment:
I want the route to be `GET /processed_documents/count` because it is more REST-like compared to `GET /count`

Reply:
Yes I thought so. This API is not really REST isn't it. It is more like a lot RPCs. Nice thing about having the prefixes there and no other route inside, is that we can see in build-server.ts where all the routes are registered